tests/integration/test_users.py

```diff
@@ -140,6 +140,17 @@ def test_user_roles(client, auth_headers):
     assert response.status_code == 200
 
 
+def test_user_no_roles(client, auth_headers, test_database):
```

Comment:
One final change I'd like on this test is to switch it over to the ~~`admin_header`~~ `valid_header` fixture.  

It's hard to tell from looking through the codebase, but we're trying to migrate to these individual headers rather than the `auth_header` fixture, since `auth_header` hits (and thus tests) the login endpoint every single time.
 
```suggestion
def test_user_no_roles(client, test_database, valid_header):
```

You'll also need to change line 148 over to `headers=admin_header`

Reply:
> One final change I'd like on this test is to switch it over to the admin_header fixture

Do you mean `valid_header` ?

I would suggest that we should create a new test file to put these tests in since there is a new class that is being created. Probably put the new class in a separate file too to keep things organized.
`app/resources/users/pending_users.py` <- possible location of new class
`tests/integration/users/test_pending_users.py` <- possible location of test

I would suggest taking a look at the class based tests that use the `empty_test_db` as the pattern for emulating anything new that is written.

Reply:
> Do you mean valid_header ?

Yes! Thanks for catching that @NickSchimek! 
I saw `auth_headers["admin"]` being used and I immediately wrote `admin_header`, but you're right &mdash; this endpoint doesn't specifically require admin authorization. I've changed my above comment to reflect that.